core-web/apps/dotcms-ui/src/app/api/services/dot-custom-event-handler/dot-custom-event-handler.service.ts

```diff
@@ -41,23 +46,37 @@ export class DotCustomEventHandlerService {
         private dotWorkflowEventHandlerService: DotWorkflowEventHandlerService,
         private dotGenerateSecurePasswordService: DotGenerateSecurePasswordService,
         private dotEventsService: DotEventsService,
-        private dotLicenseService: DotLicenseService
+        private dotLicenseService: DotLicenseService,
+        private router: Router,
+        private dotPropertiesService: DotPropertiesService
     ) {
-        if (!this.handlers) {
-            this.handlers = {
-                'edit-page': this.goToEditPage.bind(this),
-                'edit-contentlet': this.editContentlet.bind(this),
-                'edit-task': this.editTask.bind(this),
-                'create-contentlet': this.createContentlet.bind(this),
-                'company-info-updated': this.setPersonalization.bind(this),
-                'push-publish': this.pushPublishDialog.bind(this),
-                'download-bundle': this.downloadBundleDialog.bind(this),
-                'workflow-wizard': this.executeWorkflowWizard.bind(this),
-                'generate-secure-password': this.generateSecurePassword.bind(this),
-                'compare-contentlet': this.openCompareDialog.bind(this),
-                'license-changed': this.updateLicense.bind(this)
-            };
-        }
+        this.dotPropertiesService
+            .getKey(FeaturedFlags.FEATURE_FLAG_CONTENT_EDITOR2_ENABLE)
+            .pipe(map((value) => value === 'true'))
+            .subscribe((contentEditorFeatureFlag) => {
+                if (!this.handlers) {
+                    this.handlers = {
```

Comment:
Do we have a interface for this?

Reply:
No but I typed it as `Record<string, (event: CustomEvent) => void>`